pkg/providers/cf/provider.go

```diff
@@ -0,0 +1,363 @@
+				HealthCheckHTTPEndpoint:      *proc.HealthCheck.Data.Endpoint,
+				HealthCheckInvocationTimeout: uint(*proc.HealthCheck.Data.InvocationTimeout),
+				Instances:                    &procInstances,
+				LogRateLimitPerSecond:        fmt.Sprintf("%d", proc.LogRateLimitInBytesPerSecond),
+				Memory:                       fmt.Sprintf("%dMB", proc.MemoryInMB),
+				// Timeout
+				ReadinessHealthCheckType:         cfTypes.AppHealthCheckType(proc.ReadinessCheck.Type),
+				ReadinessHealthCheckHttpEndpoint: *proc.ReadinessCheck.Data.Endpoint,
+				ReadinessHealthInvocationTimeout: uint(*proc.ReadinessCheck.Data.InvocationTimeout),
+				ReadinessHealthCheckInterval:     uint(*proc.ReadinessCheck.Data.Interval),
+				Lifecycle:                        string(app.Lifecycle.Type),
+			})
+		}
+		routeOpts := client.NewRouteListOptions()
+		routeOpts.SpaceGUIDs.EqualTo(remoteSpace.GUID)
+		routes, err := cfClient.Routes.ListForAppAll(ctx, app.GUID, routeOpts)
+		if err != nil {
+			return nil, fmt.Errorf("error getting processes: %v", err)
+		}
+		appRoutes := cfTypes.AppManifestRoutes{}
+		for _, r := range routes {
+			appRoutes = append(appRoutes, cfTypes.AppManifestRoute{
+				Route:    r.URL,
+				Protocol: cfTypes.AppRouteProtocol(r.Protocol),
+				// TODO: Options: loadbalancing?
+			})
+		}
+
+		// FIXME: uncomment this when we know how to handle them
+		// allBuildpacks, err := c.cfClient.Buildpacks.ListAll(ctx, nil)
+		// if err != nil {
+		// 	return fmt.Errorf("error getting buildpacks: %v", err)
+		// }
+
+		// FIXME: uncomment this when we know how to handle them
+		// stacks, err := c.cfClient.Stacks.ListAppsOnStackAll(ctx, nil)
+		// if err != nil {
+		// 	return fmt.Errorf("error getting buildpacks: %v", err)
+		// }
+
+		// FIXME: uncomment this when we know how to handle them
+		// serviceOfferingOpts := client.NewServiceOfferingListOptions()
+		// serviceOfferingOpts.SpaceGUIDs.EqualTo(remoteSpace.GUID)
+		// serviceOfferings, err := c.cfClient.ServiceOfferings.ListAll(ctx, serviceOfferingOpts)
+		// if err != nil {
+		// 	return fmt.Errorf("error getting service offerings: %v", err)
+		// }
+
+		appManifest := cfTypes.AppManifest{
+			Name: app.Name,
+			Env:  appEnv.EnvVars, //TODO: Running, staging, appEnvVar, SystemEnvVar
+			Metadata: &cfTypes.AppMetadata{
+				Labels:      app.Metadata.Labels,
+				Annotations: app.Metadata.Annotations,
+			},
+			Processes: &appProcesses,
+			Routes:    &appRoutes,
+			// AppManifestProcess
+			// Buildpacks  //<--- There is the buildpack list but not per app
+			// RandomRoute
+			// NoRoute
+			// Services
+			// Sidecars
+			// Stack
+		}
+		if helpers.WriteToYAMLFile(appManifest, fmt.Sprintf("manifest_%s_%s.yaml", spaceName, appManifest.Name)) != nil {
+			return nil, fmt.Errorf("error writing manifest to file: %v", err)
+		}
+		appManifests = append(appManifests, appManifest)
+	}
+	return appManifests, nil
+
+}
+
+type ValidationErrorList []error
```

Comment:
You can use `error` and then join them:
```
var errors error

errors.Join(errors,<new error>)
```

https://lukas.zapletalovi.com/posts/2022/wrapping-multiple-errors/